test/ETO/ETOCommitment.js

```diff
@@ -416,18 +428,36 @@ contract("ETOCommitment", ([, admin, company, nominee, ...investors]) => {
 
     it("with changing ether price during ETO", async () => {
       await skipTimeTo(publicStartDate.add(1));
+      // set some round ETH price to minimize rounding errors in the contract
+      await gasExchange.setExchangeRate(etherToken.address, euroToken.address, Q18, {
+        from: admin,
+      });
       const tokenprice = tokenTermsDict.TOKEN_PRICE_EUR_ULPS;
-      const ticket = Q18.mul(107.61721).add(1);
+      const ticket = Q18.mul("1070.61721").add(1);
       await investAmount(investors[1], ticket, "ETH", tokenprice);
-      const newPrice = Q18.mul(defEthPrice.mul(2));
+      // set double price
+      const newPrice = Q18.mul(2);
       await gasExchange.setExchangeRate(etherToken.address, euroToken.address, newPrice, {
         from: admin,
       });
       await investAmount(investors[2], ticket, "ETH", tokenprice);
       // token price increased twice
       const i1ticket = await etoCommitment.investorTicket(investors[1]);
       const i2ticket = await etoCommitment.investorTicket(investors[2]);
-      expect(i1ticket[2].mul(2)).to.be.bignumber.eq(i2ticket[2]);
+      // but it does not mean that previous ticket amount is doubled due to rounding once, not twice
+      // i1ticket[2].mul(2) != i2ticket[2]
+      expect(
+        i1ticket[2]
+          .mul(2)
+          .sub(i2ticket[2])
+          .abs(),
+      ).to.be.bignumber.lt(2);
+      // this works because we set 1 eth == 2 eur
+      const tokensI2 = ticket
+        .mul(2)
+        .mul(getTokenPower())
+        .divToInt(tokenprice);
+      expect(i2ticket[2]).to.be.bignumber.eq(tokensI2);
```

Comment:
isnt' it 1 euro = 2 eth?
> .setExchangeRate(etherToken.address, euroToken.address ,2...




so in this test, we really verify that 

amountOfTokens * 2

Reply:
@rudolfix I think this test is off. I think if we want to double the ehterprice the setExchangeRate-params need to be switched. 


Also I don't understand why there should be a ```.mul(2)``` here...

Reply:
I increased price of ETH by 100% so I can buy twice equity tokens for the same ETH amount. price of the token is in EUR.

and the last thing is indeed a trick. here's a comment I added
```
// this works because we set 1 eth == 2 eur so ticketEth.mul(2) is in fact currency conversion to EUR
      // on which amount of token is calculated
```